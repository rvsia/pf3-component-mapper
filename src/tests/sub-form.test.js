import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SubForm from '../form-fields/sub-form';

describe('<SubForm />', () => {
  let initialProps;

  beforeEach(() => {
    initialProps = {
      formOptions: {
        renderForm: ({ name }) => <div key={ name }>Form item</div>,
      },
      fields: [],
    };
  });

  it('should render sub form with title and description', () => {
    const wrapper = shallow(<SubForm title="Foo" description="Bar" { ...initialProps } />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render sub form with title', () => {
    const wrapper = shallow(<SubForm title="Foo" { ...initialProps } />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render sub form with description', () => {
    const wrapper = shallow(<SubForm description="Bar" { ...initialProps } />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
