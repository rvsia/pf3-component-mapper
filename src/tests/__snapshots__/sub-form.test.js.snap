// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubForm /> should render sub form with description 1`] = `
<Fragment>
  <p>
    Bar
  </p>
  <div>
    Form item
  </div>
</Fragment>
`;

exports[`<SubForm /> should render sub form with title 1`] = `
<Fragment>
  <h3>
    Foo
  </h3>
  <div>
    Form item
  </div>
</Fragment>
`;

exports[`<SubForm /> should render sub form with title and description 1`] = `
<Fragment>
  <h3>
    Foo
  </h3>
  <p>
    Bar
  </p>
  <div>
    Form item
  </div>
</Fragment>
`;
